src/abi.ts

```diff
@@ -91,9 +111,73 @@ export class FunctionCall {
     return this.toScript().toHex();
   }
 
+  
+
+  genLaunch(txContext?: TxContext): string{
+
+
+    const constructorParams:SupportedParamType[] = this.contract.scriptedConstructor.params;
+
+      const entryMethodParams:SupportedParamType[] = this.params;
+      const entryMethod: string = this.methodName;
+      const name =  `Debug ${Object.getPrototypeOf(this.contract).constructor.contractName}`;
+      const program = `${Object.getPrototypeOf(this.contract).constructor.file}`;
+
+      let debugConfig: DebugConfiguration = {
+        type: "scrypt",
+        request: "launch",
+        name: name,
+        program: program,
+        constructorParams: constructorParams,
+        entryMethod: entryMethod,
+        entryMethodParams: entryMethodParams
+      }
+
+
+      const txCtx: TxContext = Object.assign({}, this.contract.txContext || {}, txContext || {});
+
+      let debugTxContext = {};
+
+      if(!isEmpty(txCtx)) {
+
+        const tx = txCtx.tx || '';
+        const inputIndex = txCtx.inputIndex || 0;
+        const inputSatoshis = txCtx.inputSatoshis || 0;
+        Object.assign(debugTxContext, {hex: tx.toString(), inputIndex,  inputSatoshis});
+      }
+
+      if(this.contract.dataPart) {
+        Object.assign(debugTxContext, {opReturn: this.contract.dataPart.toASM()});
+      }
+
+      if(!isEmpty(debugTxContext)) {
+        Object.assign(debugConfig, {txContext: debugTxContext});
+      }
+
+      let launch: DebugLaunch = {
+        version: "0.2.0",
```

Comment:
version also could be fixed to '0.2.0' in interface declaration.

Reply:
Fixed